File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> numset = new HashSet<>();

        for (int n : nums) numset.add(n);

        int res = 0;

        for (int n : numset) {
            if (!numset.contains(n-1)) {
                int cur = 1;
                while (numset.contains(n + cur)) cur++;
                res = Math.max(cur, res);
            }

        }

        return res;
    }
}

File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        res = []
        cur = []
        use = [False] * len(nums)

        def dfs():
            if len(cur) == len(nums):
                res.append(cur[:])
                return

            for i in range(len(nums)):
                if not use[i]:
                    cur.append(nums[i])
                    use[i] = True
                    dfs()
                    cur.pop()
                    use[i] = False

        dfs()
        return res